### PR TITLE
BLUE-1236 update terraform for geo points and binaries (attributes)

### PR DESCRIPTION
**Pull request by coutinol**

### CODE REVIEW

File: helper/general_objects/schemas.go [hunk 1]
```diff
@@ -189,6 +189,130 @@ var PageableSchema = map[string]*schema.Schema{
 	},
 }
 
+var geoPointFieldsDefSchema = map[string]*schema.Schema{
```
Comment (coutinol) [inline helper/general_objects/schemas.go:192]: not sure if it is a good idea to factorize geoPointFieldsDefSchema &  geoPointFieldsSchema (they differ like this: for geopoints used in DefinitionAttributes, the different fields have a defaultValue attribute, whereas the fields of the geopoints used in ValueAttributes have a value attribute)

File: docs/data-sources/activity_definitions.md [hunk 2]
```diff
@@ -132,6 +133,27 @@ Read-Only:
 - `unit_id` (String)
 
 
+<a id="nestedobjatt--content--argument_definitions--attributes--fields"></a>
```
Comment (coutinol) [inline docs/data-sources/activity_definitions.md:136]: I don't know what I'm doing  :)

File: testing/activities/activity_definitions/main.tf [hunk 3]
```diff
@@ -93,6 +93,32 @@ resource "leanspace_activity_definitions" "test" {
...
+        }
+        latitude {
+          value = 48.5
+        }
+        longitude {
+          value = 7.7
+        }
+      }
+    }
+  }
```
Comment (coutinol) [inline testing/activities/activity_definitions/main.tf:121]: I have also tested that terraform init fails when specifying data types not supported by activity definitions (same for command definitions) - I don't know what else I can test; any idea is welcome!

**Comment by coutinol** [comment]
also adding @FlorianLacour & @leanspace-thomas as I have fixed terraform for command definitions as well

File: docs/data-sources/activity_definitions.md (hunk 2 above)
Comment (leanspace-thomas) [reply docs/data-sources/activity_definitions.md:136]: I suppose it was auto-generated, right?

Comment (coutinol) [reply docs/data-sources/activity_definitions.md:136]: ah I think you're right, we can generate this - forgot about it, thanks - will remove my manual changes then and generate it

Comment (coutinol) [reply docs/data-sources/activity_definitions.md]: actually the doc is generated by an automated workflow, right?

**Comment by coutinol** [comment]
we still have some duplicated code in the general objects parser, but I think it is acceptable, as otherwise we might end up with factorizing code, that will make it more complex in the overall -> this is what was followed so far for such cases (in the scan of main branch, we actually have 9.84% of duplicated code. @jerome-leanspace shouldn't we update the sonar rule?

File: services/activities/activity_definitions/schemas.go [hunk 4]
```diff
@@ -114,8 +114,12 @@ var argumentDefinitionSchema = map[string]*schema.Schema{
 		MinItems: 1,
 		MaxItems: 1,
 		Elem: &schema.Resource{
-			Schema: general_objects.DefinitionAttributeSchema([]string{}, nil, false),
-		},
+			Schema: general_objects.DefinitionAttributeSchema(
+				[]string{"STRUCTURE", "TLE"}, // attribute types not allowed in command definition attributes
```
Comment (elodietinland) [inline services/activities/activity_definitions/schemas.go:118]:
```suggestion
				[]string{"STRUCTURE", "TLE"}, // attribute types not allowed in activity definition attributes
```
?

**Review by elodietinland — APPROVED** [review]

File: docs/data-sources/activity_definitions.md (hunk 2 above)
Comment (elodietinland) [reply docs/data-sources/activity_definitions.md]: It looks like it's done [here ](https://github.com/leanspace/terraform-provider-leanspace/blob/b5592ee3a408ccd77bacfb2a8697edf6920c1aee/.github/workflows/release.yml#L106)during the release

**Review by FlorianLacour — APPROVED** [review]

**Pull request merged**